core/src/main/java/denominator/mock/MockAllProfileResourceRecordSetApi.java

```diff
@@ -83,37 +88,28 @@ public void put(ResourceRecordSet<?> rrset) {
 
   @Override
   public void deleteByNameTypeAndQualifier(String name, String type, String qualifier) {
-    ResourceRecordSet<?> rrsMatch = getByNameTypeAndQualifier(name, type, qualifier);
-    if (rrsMatch != null) {
-      records.remove(rrsMatch);
+    Collection<ResourceRecordSet<?>> records = records();
+    synchronized (records) {
+      ResourceRecordSet<?> rrsMatch = getByNameTypeAndQualifier(records, name, type, qualifier);
+      if (rrsMatch != null) {
+        records.remove(rrsMatch);
+      }
     }
   }
 
   @Override
   public void deleteByNameAndType(String name, String type) {
-    for (Iterator<ResourceRecordSet<?>> it = iterateByNameAndType(name, type); it.hasNext(); ) {
-      records.remove(it.next());
+    Collection<ResourceRecordSet<?>> records = records();
+    synchronized (records) {
+      for (Iterator<ResourceRecordSet<?>> it = iterateByNameAndType(name, type); it.hasNext(); ) {
+        records.remove(it.next());
+      }
     }
   }
 
-  static class Factory implements denominator.AllProfileResourceRecordSetApi.Factory {
-
-    private final Provider provider;
-    private final Map<Zone, SortedSet<ResourceRecordSet<?>>> records;
-
-    // unbound wildcards are not currently injectable in dagger
-    @SuppressWarnings({"rawtypes", "unchecked"})
-    @Inject
-    Factory(Provider provider, Map<Zone, SortedSet<ResourceRecordSet>> records) {
```

Comment:
mock doesn't use DI to inject factories anymore, so no more confusing comments about unbound wildcards